apps/api/src/donations/donations.service.ts

```diff
@@ -619,31 +680,36 @@ export class DonationsService {
    * @param updatePaymentDto
    * @returns
    */
-  async update(id: string, updatePaymentDto: UpdatePaymentDto): Promise<Donation> {
+  async update(id: string, updatePaymentDto: UpdatePaymentDto): Promise<Payment> {
     try {
       // execute the below in prisma transaction
       return await this.prisma.$transaction(async (tx) => {
-        const currentDonation = await tx.donation.findFirst({
+        const currentDonation = await tx.payment.findFirst({
```

Comment:
the variable here should be currentPayment

Reply:
Done.